apptest.go

```diff
@@ -214,25 +281,33 @@ func (a *AppSetup) waitForDeployedApps(ctx context.Context, apps []App) error {
 }
 
 func (a *AppSetup) waitForDeployedApp(ctx context.Context, appName string) error {
+	var err error
+
 	a.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring %#q app CR is %#q", appName, deployedStatus))
 
+	var app v1alpha1.App
+
 	o := func() error {
-		app, err := a.k8sClient.G8sClient().ApplicationV1alpha1().Apps(namespace).Get(ctx, appName, metav1.GetOptions{})
+		err = a.ctrlClient.Get(
+			ctx,
+			types.NamespacedName{Name: appName, Namespace: namespace},
+			&app)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 		if app.Status.Release.Status != deployedStatus {
 			return microerror.Maskf(executionFailedError, "waiting for %#q, current %#q", deployedStatus, app.Status.Release.Status)
 		}
+
 		return nil
 	}
 
 	n := func(err error, t time.Duration) {
 		a.logger.Log("level", "debug", "message", fmt.Sprintf("failed to get app CR status '%s': retrying in %s", deployedStatus, t), "stack", fmt.Sprintf("%v", err))
 	}
 
-	b := backoff.NewConstant(20*time.Minute, 60*time.Second)
-	err := backoff.RetryNotify(o, b, n)
+	b := backoff.NewConstant(20*time.Minute, 30*time.Second)
```

Comment:
Reducing the backoff. Its still slow because it needs the resync period.

But the app CRs use "in cluster" so we can use the status webhook. 🎉

Reply:
Only if app-operator & chart-operator are in their latest version.